core/common/src/main/scala/org/typelevel/otel4s/TextMapSetter.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+trait TextMapSetter[A] {
+  def unsafeSet(carrier: A, key: String, value: String): Unit
```

Comment:
The spec mandates `set` and mutability of carrier.  The unsafety is efficiency, knowing the `TextMapPropagator` wraps it in `F.delay`.